File: initial/src/main/java/com/example/restservice/RestServiceApplication.java
package com.example.restservice;

// SpringApplication: Spring Bootアプリケーションを起動するためのクラス
import org.springframework.boot.SpringApplication;
// SpringBootApplication: Spring Bootの自動設定を有効にするアノテーション
import org.springframework.boot.autoconfigure.SpringBootApplication;

/*
 * RestServiceApplication.java - アプリケーションのエントリーポイント（起動点）
 *
 * 【解説】
 * これはSpring Bootアプリケーションの「初期状態（initial）」です。
 * まだREST APIのエンドポイント（Greeting, GreetingController）は実装されていません。
 * complete/ ディレクトリに完成版があります。
 *
 * @SpringBootApplication は、以下の3つのアノテーションを1つにまとめたものです：
 *   1. @Configuration      : このクラスがBean（部品）の定義を含むことを示す
 *   2. @EnableAutoConfiguration : Spring Bootが自動的に設定を行う
 *   3. @ComponentScan      : このパッケージ以下のクラスを自動検出・登録する
 *
 * main() メソッドがJavaプログラムのエントリーポイント（最初に実行される場所）です。
 * SpringApplication.run() を呼ぶことで、組み込みWebサーバーが起動します。
 */
@SpringBootApplication
public class RestServiceApplication {

  // Javaプログラムのエントリーポイント
  public static void main(String[] args) {
    // Spring Bootアプリケーションを起動
    SpringApplication.run(RestServiceApplication.class, args);
  }

}

File: complete-kotlin/src/main/kotlin/com/example/restservice/RestServiceApplication.kt
package com.example.restservice

// SpringBootApplication: Spring Bootの自動設定を有効にするアノテーション
import org.springframework.boot.autoconfigure.SpringBootApplication
// runApplication: Kotlinの拡張関数で、SpringApplication.run() をより簡潔に呼べる
import org.springframework.boot.runApplication

/*
 * RestServiceApplication.kt - アプリケーションのエントリーポイント（起動点）（Kotlin版）
 *
 * 【解説】
 * @SpringBootApplication: Spring Bootの中核となるアノテーション。
 *   以下の3つの機能を1つにまとめています：
 *   1. @Configuration      : Bean（部品）の定義を含むクラスであることを示す
 *   2. @EnableAutoConfiguration : ライブラリに基づく自動設定を有効にする
 *   3. @ComponentScan      : このパッケージ以下のクラスを自動検出・登録する
 *
 * Kotlinでは、クラス本体が空の場合は {} を省略できます。
 * また、main関数はクラスの外（トップレベル）に定義します。
 * これはJavaと異なり、Kotlinではクラスに属さない関数を書けるためです。
 */
@SpringBootApplication
class RestServiceApplication

// Kotlinプログラムのエントリーポイント（最初に実行される関数）
// args: コマンドライン引数の配列
// *args: スプレッド演算子。配列を個別の引数として展開して渡す
fun main(args: Array<String>) {
  // runApplication はKotlin用のヘルパー関数。
  // <RestServiceApplication> は型パラメータ（ジェネリクス）で、起動するクラスを指定。
  runApplication<RestServiceApplication>(*args)
}
